### PR TITLE
fix doc formatting errors

### PR DESCRIPTION
**Pull request by jabielecki**
Fix obvious documentation formatting errors:

- Use `* ` prefix for rendering lists (instead of `- ` which has no effect).
- Add empty lines as separators before and after lists.
- Fix punctuation where it was incoherent.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: products/compute/api.yaml [hunk 1]
```diff
@@ -1801,7 +1803,7 @@ objects:
       - !ruby/object:Api::Type::NestedObject
         name: 'consistentHash'
         description: |
-          Consistent Hash-based load balancing can be used to provide soft session
+          Consistent hash-based load balancing can be used to provide soft session
```
Comment (rileykarson) [inline products/compute/api.yaml:1806]: This one comes from https://cloud.google.com/compute/docs/reference/rest/v1/backendServices, so I think we want to preserve it.
```suggestion
          Consistent Hash-based load balancing can be used to provide soft session
```

**Review by rileykarson — COMMENTED** [review]
Thanks @jabielecki! Just one small nitpick, I think.

File: products/compute/api.yaml (hunk 1 above)
Comment (jabielecki) [reply products/compute/api.yaml]: Done. :raised_eyebrow:

**Comment by jabielecki** [comment]
> Thanks @jabielecki! Just one small nitpick, I think.

Ready for re-review.

**Review by rileykarson — APPROVED** [review]
LGTM- thanks!

**Pull request merged**